app/src/main/java/com/example/mvp/ui/signin/SignInPresenter.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.mvp.ui.signin
+
+import com.example.mvp.data.model.user
+
+class SignInPresenter: SignInContract.Presenter {
+
+    private var mView : SignInContract.View? = null
+    val listUser = mutableListOf<user>()
+
+
+    fun setView(view: SignInContract.View){
+        mView = view
+    }
+
+    override fun handleSignIn( userName: String, passWord : String) {
+        for(userX in listUser!!) {
+            if (userName == userX.userName && passWord == userX.passWord) {
+                mView?.signInSuccess(userX)
+                return
+            }
+        }
+        mView?.signInFailure("User Name or Password")
```

Comment:
```suggestion
        listUser
            .firstOrNull { userX ->
                userName == userX.userName && passWord == userX.passWord
            }
            ?.let(view::signInSuccess)
            ?: view.signInFailure(/*...*/)
```
Ô tham khảo cách viết nâng cao hơn nhé